docs/multi-tenancy/README.md

```diff
@@ -0,0 +1,53 @@
+# Multi-tenant Minio Deployment Guide [![Slack](https://slack.minio.io/slack?type=svg)](https://slack.minio.io) [![Go Report Card](https://goreportcard.com/badge/minio/minio)](https://goreportcard.com/report/minio/minio) [![Docker Pulls](https://img.shields.io/docker/pulls/minio/minio.svg?maxAge=604800)](https://hub.docker.com/r/minio/minio/) [![codecov](https://codecov.io/gh/minio/minio/branch/master/graph/badge.svg)](https://codecov.io/gh/minio/minio)
+
+## Standalone Deployment
+To host multiple tenants on a single machine, run one Minio server per tenant with dedicated HTTPS port, config and data directory.  
+
+#### Example 1 : Single host, single drive
+
+This example hosts 3 tenants on a single drive.
+```sh
+minio --config-dir ~/tenant1 server --address :9001 /disk1/data/tenant1
+minio --config-dir ~/tenant2 server --address :9002 /disk1/data/tenant2
+minio --config-dir ~/tenant3 server --address :9003 /disk1/data/tenant3
+```
+
+![Example-1](https://raw.githubusercontent.com/minio/minio/master/docs/screenshots/Example-1.png)
+
+#### Example 2 : Single host, multiple drives (erasure code)
+
+This example hosts 3 tenants on multiple drives.
+```sh
+minio --config-dir ~/tenant1 server --address :9001 /disk1/data/tenant1 /disk2/data/tenant1 /disk3/data/tenant1 /disk4/data/tenant1
+minio --config-dir ~/tenant2 server --address :9002 /disk1/data/tenant2 /disk2/data/tenant2 /disk3/data/tenant2 /disk4/data/tenant2
+minio --config-dir ~/tenant3 server --address :9003 /disk1/data/tenant3 /disk2/data/tenant3 /disk3/data/tenant3 /disk4/data/tenant3
+```
+![Example-2](https://raw.githubusercontent.com/minio/minio/master/docs/screenshots/Example-2.png)
+
+## Distributed Deployment
+To host multiple tenants in a distributed environment, run several distributed Minio instances concurrently.  
+
+#### Example 1 : Multiple host, multiple drives (erasure code)
+
+This example hosts 3 tenants on a 4 node distributed setup. Execute the following command on all the four nodes. 
+
+```sh
+export MINIO_ACCESS_KEY=<TENANT1_ACCESS_KEY>
+export MINIO_SECRET_KEY=<TENANT1_SECRET_KEY>
```

Comment:
> Presigned URLs should be part of multi-tenancy document. Because for application developers, all they need is presigned URLs to support multi-tenancy in their application. For an application with a million users where each user will have 10-15 objects, we don't want to suggest that they need to deploy one minio per user.

As per the discussion with @abperiasamy we can start off with this and then add another doc that explains ways to access Minio i.e. - Presigned URLs, Anonymous access and via credentials.